pkg/kapp/app/recorded_app.go

```diff
@@ -105,8 +105,13 @@ func (a *RecordedApp) UpdateUsedGVsAndGKs(gvs []schema.GroupVersion, gks []schem
 		}
 	}
 
+	sort.Slice(uniqGVs, func(i int, j int) bool {
+		return uniqGVs[i].Group+uniqGVs[i].Version < uniqGVs[j].Group+uniqGVs[j].Version
+	})
+
 	gksByGK := map[schema.GroupKind]struct{}{}
-	var uniqGKs []schema.GroupKind
+	// Initialize with an empty slice so that we do not remove `usedGKs` from meta configmap
+	uniqGKs := []schema.GroupKind{}
```

Comment:
Do we want to make the same change for GVs as well?

Reply:
We don't have a similar check on GVs, so it won't really be helpful.
Fir GKs, if it's nil, then we assume that it's an old app and we don't scope list calls to used GKs.

Reply:
Gotcha, makes sense